core/ipc/rpc/camel-impl/src/main/java/org/opennms/core/rpc/camel/CamelRpcServerRouteManager.java

```diff
@@ -77,7 +76,11 @@ private DynamicRpcRouteBuilder(CamelContext context, MinionIdentity identity, Rp
         }
 
         public String getRouteId() {
-            return "RPC.Server." + module.getId();
+            return "RPC.Server." + module.getId() + "-" + Math.abs(module.hashCode());
```

Comment:
For that reason, I feel that we should revert the change on this line. Loading the same modules multiple times is not a problem during the normal run-time, and if it does reoccur in the tests, we can still isolate it using the log message you added.